packages/flutter/lib/src/cupertino/text_field.dart

```diff
@@ -125,6 +125,14 @@ class _CupertinoTextFieldSelectionGestureDetectorBuilder extends TextSelectionGe
     _state._requestKeyboard();
     super.onDragSelectionEnd(details);
   }
+
+  @override
+  void onDragSelectionUpdate(TapDragUpdateDetails details) {
+    if (!_state._effectiveFocusNode.hasFocus) {
```

Comment:
This works in practice but not in some tests below (without autofocusing the field first).  In onDragSelectionStart, the field is focused with `requestFocus`, but that may take an additional frame to be set in `hasFocus`.  In the tests, onDragSelectionUpdate is called before `hasFocus` is true.